frontend/src/app/main/ui/workspace/tokens/token_pill.cljs

```diff
@@ -37,21 +159,46 @@
                           full-applied
                           "token-status-full"
                           :else
-                          "token-status-non-applied")]
+                          "token-status-non-applied")
+
+        on-context-menu
+        (mf/use-fn
+         (mf/deps can-edit? on-context-menu)
+         (fn [e]
+           (dom/stop-propagation e)
+           (when can-edit?
+             (on-context-menu e))))
+
+        on-click
+        (mf/use-fn
+         (mf/deps errors? on-click)
+         (fn [event]
+           (dom/stop-propagation event)
+           (when (and can-edit? (not (seq errors)) on-click)
+             (on-click event))))
+
+        on-hover
+        (mf/use-fn
+         (mf/deps selected-shapes viewer)
+         (fn [event]
+           (let [node (dom/get-current-target event)
+                 calculate-tooltip (token-pill-tooltip theme-token viewer (first selected-shapes) token-type-props token half-applied)]
```

Comment:
I think this should be called title, `calculate-tooltip`  looks like a function name